src/gordon_gcp/plugins/enricher.py

```diff
@@ -127,19 +127,28 @@ def __init__(self, config, http_client, success_channel, error_channel):
         self._http_client = http_client
         self._logger = logging.getLogger('')
 
+    async def _check_instance_data(self, resource_name, instance_data,
+                                   msg_logger):
```

Comment:
Can you give an example? An error is already logged if it fails to load, so  I don't see how a subclass would need to log anything else.  I also think this method should be sync since it should just be checking the response body. No other requests should happen here.

I am arguing against removing so much stuff because I am afraid we are making a more complicated and harder to follow API based on potential use cases, which will make it harder to use and reason about in the use cases that we know exist.